src/dotnet/APIView/APIViewWeb/Managers/ReviewManager.cs

```diff
@@ -461,63 +429,86 @@ private async Task MarkAssociatedReviewsForNamespaceReview(List<string> associat
         }
 
         /// <summary>
-        /// Find related language reviews by looking up pull request numbers from TypeSpec review
+        /// Find related language reviews by looking up pull request number from specific TypeSpec revision
         /// </summary>
         /// <param name="typeSpecReviewId">The TypeSpec review ID</param>
-        /// <returns>List of review IDs for related language reviews</returns>
-        private async Task<List<string>> FindRelatedReviewsByPullRequestAsync(string typeSpecReviewId)
+        /// <param name="revisionId">The specific revision ID to get pull request from</param>
+        /// <returns>List of related language reviews</returns>
+        private async Task<List<ReviewListItemModel>> FindRelatedReviewsByPullRequestAsync(string typeSpecReviewId, string revisionId)
         {
-            var relatedReviewIds = new List<string>();
+            var relatedReviews = new List<ReviewListItemModel>();
             
             try
             {
-                // Get all revisions for the TypeSpec review to find pull request numbers
-                var revisions = await _apiRevisionsManager.GetAPIRevisionsAsync(typeSpecReviewId);
-                var pullRequestNumbers = revisions
-                    .Where(r => r.PullRequestNo.HasValue && r.PullRequestNo.Value > 0)
-                    .Select(r => r.PullRequestNo.Value)
-                    .Distinct()
-                    .ToList();
-
-                // For each pull request number, find related reviews
-                foreach (var prNumber in pullRequestNumbers)
+                // Get the specific revision to find its pull request number
+                var revision = await _apiRevisionsManager.GetAPIRevisionAsync(revisionId);
+
+                // TODO FIX: current db is showing null for PullRequestNo in APIRevision
+                int? prNumber = revision?.PullRequestNo;
+                if (!prNumber.HasValue && !string.IsNullOrEmpty(revision?.Label))
```

Comment:
This logic is incorrect. We should fetch Pull request revision details from Pull Requests DB which has pull request manager. We must not use label to decide the PR.